src/v0/destinations/intercom_v2/utils.js

```diff
@@ -0,0 +1,332 @@
+ * 401 - code: unauthorized, message: Access Token Invalid
+ * 404 - code: company_not_found, message: Company Not Found
+ * @param {*} message
+ * @param {*} processedResponse
+ */
+const intercomErrorHandler = (message, processedResponse) => {
+  const errorMessages = JSON.stringify(processedResponse.response);
+  if (processedResponse.status === 400) {
+    throw new InstrumentationError(`${message} : ${errorMessages}`);
+  }
+  if (processedResponse.status === 401) {
+    throw new InvalidAuthTokenError(message, 400, errorMessages);
+  }
+  if (processedResponse.status === 404) {
+    throw new InstrumentationError(`${message} : ${errorMessages}`);
+  }
+  throw new NetworkError(
+    `${message} : ${errorMessages}`,
+    processedResponse.status,
+    {
+      [tags]: getDynamicErrorType(processedResponse.status),
+    },
+    processedResponse,
+  );
+};
+
+const getHeaders = (metadata) => ({
+  Authorization: `Bearer ${getAccessToken(metadata, 'accessToken')}`,
+  Accept: JSON_MIME_TYPE,
+  'Content-Type': JSON_MIME_TYPE,
+  'Intercom-Version': ApiVersions.v2,
+});
+
+const getBaseEndpoint = (destination) => {
+  const { apiServer } = destination.Config;
+  switch (apiServer) {
+    case 'Europe':
+      return BASE_EU_ENDPOINT;
+    case 'Australia':
+      return BASE_AU_ENDPOINT;
+    default:
+      return BASE_ENDPOINT;
+  }
+};
+
+const getStatusCode = (event) => {
+  const { message } = event;
+  let statusCode = HTTP_STATUS_CODES.OK;
+  const messageType = getEventType(message);
+  if (messageType === EventType.GROUP) {
+    statusCode = HTTP_STATUS_CODES.SUPPRESS_EVENTS;
+  }
+  return statusCode;
+};
+
+const getResponse = (method, endpoint, headers, payload) => {
+  const response = defaultRequestConfig();
+  response.method = method;
+  response.endpoint = endpoint;
+  response.headers = headers;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  return response;
+};
+
+const searchContact = async (event) => {
+  const { message, destination, metadata } = event;
+  const lookupField = getLookUpField(message);
+  let lookupFieldValue = getFieldValueFromMessage(message, lookupField);
+  if (!lookupFieldValue) {
+    lookupFieldValue = message?.context?.traits?.[lookupField];
+  }
+  const data = JSON.stringify({
+    query: {
+      operator: 'AND',
+      value: [
+        {
+          field: lookupField,
+          operator: '=',
+          value: lookupFieldValue,
+        },
+      ],
+    },
+  });
+
+  const headers = getHeaders(metadata);
+  const endpoint = `${getBaseEndpoint(destination)}/${SEARCH_CONTACT_ENDPOINT}`;
+  const statTags = {
+    destType,
+    feature: 'transformation',
+    endpointPath: '/contacts/search',
+    requestMethod: 'POST',
+    module: 'router',
+    metadata,
+  };
+
+  const { processedResponse: response } = await handleHttpRequest(
+    'POST',
+    endpoint,
+    data,
+    {
+      headers,
+    },
+    statTags,
```

Comment:
Can we send the metadata to `handleHttpRequest` function calls? It will help us with logging.

Reply:
metadata is present in statTags.

Reply:
metadata coming in log. resolving this.